magento/magento2
# magento/magento2#37028: [Sales] Missing relation data when reorder

- fix createOrder method to understand that order is reordered.





### Description (*)

Change \Magento\Sales\Model\AdminOrder\Create::createOrder to understand that order is reordered, before changes part of this method didn't work because it tries to get order from session, now it looks to 'reordered' parameter that added in \Magento\Sales\Model\AdminOrder\Create::initFromOrder 

### Related Pull Requests

--

### Fixed Issues (if relevant)

1. Fixes magento/magento2#37028

### Manual testing scenarios (*)

1. Place an order
2. Reorder that order from backend
3. Check on backend order page view

Before fix:
relation_child_id and relation_child_real_id are not set in parent order row. No link is diplayed on parent order view to new order.
relation_parent_id and relation_parent_real_id are not set in child order row. No link is diplayed on child order view to parent order.

After fix:
relation_child_id and relation_child_real_id are set in parent order row. A link is diplayed on parent order view to new order.
relation_parent_id and relation_parent_real_id are set in child order row. A link is diplayed on child order view to parent order.

### Questions or comments


### Contribution checklist (*)
